From d79a161f89eebe7e5d1cf619d94ecc759e7bbdcd Mon Sep 17 00:00:00 2001
From: Julian Bouzas <julian.bouzas@collabora.com>
Date: Mon, 28 Jul 2025 15:50:39 -0400
Subject: [PATCH 1/4] m-si-audio-adapter: Configure the node ports if the item
 has been re-configured

If we re-configure the adapter with different settings than the ones from the
first configuration, we also need to configure the node ports to make sure they
are updated with the new settings.

For example, this is need for stream audio adapters that have been configured
with monitor ports, and later re-configured without monitor ports. Without this
change, since stream audio adapters never configure the node ports on activation,
the internal node will still have the monitor ports present even after disabling
them in the 2nd re-configuration.
---
 modules/module-si-audio-adapter.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/modules/module-si-audio-adapter.c b/modules/module-si-audio-adapter.c
index 995345a6..048a93e2 100644
--- a/modules/module-si-audio-adapter.c
+++ b/modules/module-si-audio-adapter.c
@@ -24,6 +24,7 @@ struct _WpSiAudioAdapter
   WpSessionItem parent;
 
   /* configuration */
+  gboolean reconfigured;
   WpNode *node;
   WpPort *port;  /* only used for passthrough or convert mode */
   gboolean no_format;
@@ -101,6 +102,7 @@ si_audio_adapter_reset (WpSessionItem * item)
   si_audio_adapter_soft_reset (self);
 
   /* reset */
+  self->reconfigured = self->node != NULL;
   g_clear_object (&self->node);
   g_clear_object (&self->port);
   self->no_format = FALSE;
@@ -611,8 +613,10 @@ si_audio_adapter_enable_active (WpSessionItem *si, WpTransition *transition)
 
   /* If device node, enum available formats and set one of them */
   if (!self->no_format && (self->is_device || self->dont_remix ||
-      !self->is_autoconnect || self->disable_dsp || self->is_unpositioned))
+      !self->is_autoconnect || self->disable_dsp || self->is_unpositioned ||
+      self->reconfigured)) {
     si_audio_adapter_configure_node (self, transition);
+  }
 
   /* Otherwise just finish activating */
   else
-- 
2.50.1

